### PR TITLE
docs: update the partition grammar

### PR DESCRIPTION
**Pull request by waynexia**
## What's Changed in this PR

Update the new partition grammar in 0.7

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

### CODE REVIEW

**Review by tisonkun — COMMENTED** [review]
Is this currently an unaligned content between en and zh version?

**Comment by waynexia** [comment]
>This change requires follow-up update in localized docs.

File: docs/v0.7/zh/contributor-guide/frontend/table-sharding.md [hunk 1]
```diff
@@ -8,32 +8,39 @@
...
+在 “RANGE COLUMNS 分区”中，每个分区仅包含表中的一部分数据，并按某些列值范围进行分组。例如，我们可以使用这样的语法在 GreptimeDB 中对表进行分区：
+
+```sql
+CREATE TABLE (...)
+PARTITION ON COLUMNS (<COLUMN LIST>) (
+    <RULE LIST>
+);
+```
+
+主要分为两部分，`PARTITION ON COLUMNS` 后跟随一个使用逗号分隔的列名列表，主要用于指定哪些列*可能会被*用于分区；`RULE LIST` 是一个包含多个分区规则的列表，每个规则都是一个分区名称和一个分区条件的组合。
```
Comment (nicecui) [inline docs/v0.7/zh/contributor-guide/frontend/table-sharding.md:20]:
```suggestion
该语法主要包含两部分：
- `PARTITION ON COLUMNS` 后跟随一个使用逗号分隔的列名列表，用于指定哪些列可能会被用于分区。
- `RULE LIST` 是一个包含多个分区规则的列表，每个规则都是一个分区名称和一个分区条件的组合。此处的表达式可使用 `=`，`!=`，`>`，`>=`，`<`，`<=`，`AND` 和 `OR`。
```

**Review by nicecui — COMMENTED** [review]
LGTM, please add the English content after committing suggestions, thank you very much!

File: docs/v0.7/zh/contributor-guide/frontend/table-sharding.md [hunk 2]
```diff
@@ -8,32 +8,39 @@
...
 我们在上面创建的 `my_table` 有 3 个分区。分区 "p0" 包含了 "a < 10" 的行；分区 "p1" 包含了 "10 <= a < 20" 的行；分区 "p2" 包含了剩下的 "a >= 20" 的所有行。
 
 ::: warning 重要
 
-1. 所有分区的范围必须严格递增，并最终以 "`MAXVALUE`" 结尾。
-2. 用于分区的列必须是主键。
-
-:::
+1. 所有分区的范围必须是不重叠的。
+2. 用于分区的列必须是在 `ON COLUMNS` 指定，但不是所有指定的列都必须用上。
```
[the remark this answers is not in the document]
Comment (waynexia) [reply docs/v0.7/zh/contributor-guide/frontend/table-sharding.md:44]: I added some explain to COLUMN LIST in https://github.com/GreptimeTeam/docs/pull/888/commits/83e186a33a82231c90650f87dfdc312bd2f43284. It's kind of allow-list.

**Review by killme2008 — COMMENTED** [review]
LGTM

**Review by killme2008 — APPROVED** [review]

**Pull request merged**